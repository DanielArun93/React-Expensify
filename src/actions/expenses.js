import uuid from 'uuid';
import database from '../firebase/firebase';

//Add expenses action generators
export const addExpense = (expense) => {
    return {
        type: 'ADDEXPENSES',
        expense
    }
}

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData

        const expense = { description, note, amount, createdAt };
        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }))
        })

    }
}

//Remove expense
export const removeExpense = ({id} = {}) => {
    return {
        type: 'REMOVEEXPENSE',
        id
    }
}

//edit expense
export const editExpense = (id, update) => {
    return {
        type: 'EDITEXPENSE',
        id,
        update
    }
}