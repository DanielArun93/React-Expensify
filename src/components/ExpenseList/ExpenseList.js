import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from '../ExpenseListItem/ExpenseListItem';
import SelectedExpenses from '../../selectors/expenses';

export const ExpenseList = (props) => {
    return (
        <div>
            {
                props.expenses.length === 0 ?
                    (
                        <p>No Expenses</p>
                    ) :
                    (
                        props.expenses.map((expense) => {
                            return <ExpenseListItem {...expense} key={expense.id} />
                        })
                    )
            }

        </div>
    )
}

// const ConnectedExpenseList = connect((state) => {
//     return {
//         expenses:state.expenses,
//         filters:state.filters
//     }
// })(ExpenseList)

const mapStateToProps = (state) => {
    return {
        expenses: SelectedExpenses(state.expenses, state.filters)

    }
}
export default connect(mapStateToProps)(ExpenseList);
