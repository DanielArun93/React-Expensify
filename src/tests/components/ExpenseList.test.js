import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList/ExpenseList';
import expenses from '../../fixtures/expenses';


test('should create ExpenseList items when expenses are available', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
})

test('should not create ExpenseList items when expenses are not available', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
})