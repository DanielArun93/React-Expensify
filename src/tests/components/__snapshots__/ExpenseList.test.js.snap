// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create ExpenseList items when expenses are available 1`] = `
<div>
  <ExpenseListItem
    amount={12000}
    createdAt={0}
    description="Rent"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={800}
    createdAt={-345600000}
    description="Card Bill"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={20000}
    createdAt={345600000}
    description="EB Bill"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should not create ExpenseList items when expenses are not available 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
